column-profiling/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -0,0 +1,121 @@
+package com.r3.conclave.sample.client;
+
+import com.r3.conclave.client.EnclaveConstraint;
+import com.r3.conclave.common.EnclaveInstanceInfo;
+import com.r3.conclave.mail.Curve25519PrivateKey;
+import com.r3.conclave.mail.EnclaveMail;
+import com.r3.conclave.mail.PostOffice;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.security.PrivateKey;
+import java.util.ArrayList;
+import java.io.*;
+import java.util.UUID;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Output;
+import com.r3.conclave.sample.dataanalysis.common.DataSetSerializer;
+import com.r3.conclave.shaded.kotlin.Pair;
+import com.r3.conclave.sample.dataanalysis.common.UserProfile;
+import com.r3.conclave.sample.dataanalysis.common.DataSet;
+
+
+public class Client {
+    private static Kryo kryo;
+
+    public static void main(String[] args) throws Exception {
+        // This is the client that will upload secrets to the enclave for processing.
+
+        kryo = new Kryo();
+        if (args.length == 0) {
+            System.err.println("Please pass the user data for processing as argument to the command line using --args=\"data fields with space separation \"");
+            return;
+        }
+        ArrayList<UserProfile> l = new ArrayList<UserProfile>() {
+        };
+        String constraint = args[0] + " " + args[1];
+        for (int i = 2; i < args.length; i = i + 4) {
+            UserProfile u = new UserProfile(args[i],
+                    Integer.parseInt(args[i + 1]),
+                    args[i + 2],
+                    args[i + 3]);
+            l.add(u);
+
+        }
```

Comment:
Consider using [picocli](https://picocli.info/) to handle the parsing of cmd line parameters. Not urgent but consider it for the future.